src/lib.rs

```diff
@@ -153,7 +180,9 @@ pub extern "C" fn rustls_client_config_builder_build(
     builder: *mut rustls_client_config_builder,
 ) -> *const rustls_client_config {
     ffi_panic_boundary_ptr! {
-        let b = unsafe { Box::from_raw(builder as *mut ClientConfig) };
+        let config: &mut ClientConfig = checked_ptr_mut!(builder, ClientConfig,
+             null::<rustls_client_config>());
+        let b = unsafe { Box::from_raw(config) };
```

Comment:
👍🏻 I like that the `unsafe` section is made smaller.

Reply:
FWIW, it's not really. The `checked_ptr_mut!` implementation is one big unsafe blocks. So now we just have two smaller unsafe blocks rather than one big one.